sjsonnet/src/sjsonnet/StdRegex.scala

```diff
@@ -4,66 +4,58 @@ import sjsonnet.Expr.Member.Visibility
 import sjsonnet.Val.Obj
 
 object StdRegex {
+  def regexPartialMatch(pos: Position, pattern: String, str: String): Val = {
+    val compiledPattern = Platform.getPatternFromCache(pattern)
+    val matcher = compiledPattern.matcher(str)
+    var returnStr: Val = null
+    val groupCount = matcher.groupCount()
+    val captures = Array.newBuilder[Val]
+    captures.sizeHint(groupCount)
+
+    while (matcher.find()) {
+      if (returnStr == null) {
+        val m = matcher.group(0)
+        if (m != null) {
+          returnStr = Val.Str(pos.noOffset, matcher.group(0))
+        } else {
+          returnStr = Val.Str(pos.noOffset, "")
+        }
+      }
```

Comment:
Once we change the `while` to an `if`, we can add an `else` and greatly simplify this entire block. For example, we can move the allocation of the captures builder into the `if` and avoid the allocation in case of no match.